pkg/build/builder/daemonless.go

```diff
@@ -305,32 +309,101 @@ func buildDaemonlessImage(sc types.SystemContext, store storage.Store, isolation
 
 func generateTransientMounts() []string {
 	mounts := []string{}
-	mounts = appendRHSMMount(mounts)
+	mounts = appendRHSMMount(defaultMountStart, mounts)
+	mounts = appendETCPKIMount(defaultMountStart, mounts)
+	mounts = appendRHRepoMount(defaultMountStart, mounts)
 	mounts = appendCATrustMount(mounts)
 	return mounts
 }
 
-func appendRHSMMount(mounts []string) []string {
-	st, err := os.Stat("/run/secrets/rhsm")
+func appendRHRepoMount(pathStart string, mounts []string) []string {
+	path := filepath.Join(pathStart, "redhat.repo")
+	st, err := os.Stat(path)
 	if err != nil {
-		log.V(0).Infof("Red Hat entitlements will not be available in this build: failed to stat RHSM secrets directory /run/secrets/rhsm: %v", err)
+		// since the presence of /run/secrets/redhat.repo is not a given, we won't log this a V(0) like
+		// we do for /run/secrets/rhsm
+		log.V(4).Infof("Failed to stat %s, falling back to the Red Hat yum repository configuration in the build's base image. Error: %v", path, err)
 		return mounts
 	}
-	if !st.IsDir() {
-		log.V(0).Infof("Red Hat entitlements will not be available in this build: rhsm secrets location /run/secrets/rhsm is not a directory")
+	if st.IsDir() {
```

Comment:
`if !st.isFile()` might be more clear here

Reply:
not 100% sure that covers symlinks

I'll try it with the new unit tests and report back

Reply:
also fyi there is no `st.IsFile()` method

the closest is `st.Mode().IsRegular()`

Reply:
Yep, that is what I was looking at: https://pkg.go.dev/gopkg.in/src-d/go-git.v4@v4.13.1/plumbing/filemode#FileMode.IsFile

Reply:
ok it works .... part of next push